powersimdata/tests/mock_grid.py

```diff
@@ -1,85 +1,23 @@
 import pandas as pd
 
-from powersimdata.input import const
+from powersimdata.input.abstract_grid import AbstractGrid
+from powersimdata.input.const import grid_const
 from powersimdata.input.grid import Grid
 from powersimdata.network.model import ModelImmutables
 
-# The index name of each data frame attribute
-indices = {
-    "sub": "sub_id",
-    "bus2sub": "bus_id",
-    "branch": "branch_id",
-    "bus": "bus_id",
-    "dcline": "dcline_id",
-    "plant": "plant_id",
-}
 
-gencost_names = {"gencost_before": "before", "gencost_after": "after"}
-storage_names = {"storage_gen": "gen", "storage_StorageData": "StorageData"}
-acceptable_keys = (
-    set(indices.keys()) | set(gencost_names.keys()) | set(storage_names.keys())
-)
-
-# The column names of each data frame attribute
-sub_columns = ["name", "interconnect_sub_id", "lat", "lon", "interconnect"]
-
-bus2sub_columns = ["sub_id", "interconnect"]
-
-branch_augment_columns = [
-    "branch_device_type",
-    "interconnect",
-    "from_zone_id",
-    "to_zone_id",
-    "from_zone_name",
-    "to_zone_name",
-    "from_lat",
-    "from_lon",
-    "to_lat",
-    "to_lon",
-]
-branch_columns = const.col_name_branch + branch_augment_columns
-
-bus_augment_columns = ["interconnect", "lat", "lon"]
-bus_columns = const.col_name_bus + bus_augment_columns
-
-dcline_augment_columns = ["from_interconnect", "to_interconnect"]
-dcline_columns = const.col_name_dcline + dcline_augment_columns
-
-gencost_augment_columns = ["interconnect"]
-gencost_columns = const.col_name_gencost + gencost_augment_columns
-
-plant_augment_columns = [
-    "type",
-    "interconnect",
-    "GenFuelCost",
-    "GenIOB",
-    "GenIOC",
-    "GenIOD",
-    "zone_id",
-    "zone_name",
-    "lat",
-    "lon",
-]
-plant_columns = const.col_name_plant + plant_augment_columns
-
-storage_columns = {
-    # The first 21 columns of plant are all that's necessary
-    "gen": plant_columns[:21],
-    "StorageData": const.col_name_storage_storagedata,
-}
-
-
-class MockGrid:
+class MockGrid(AbstractGrid):
     def __init__(self, grid_attrs=None, model="usa_tamu"):
         """Constructor.
 
         :param dict grid_attrs: dictionary of {*field_name*, *data_dict*} pairs
-            where *field_name* is the name of the data frame (sub, bus2sub,
-            branch, bus, dcline, gencost or plant) and *data_dict* a dictionary
-            in which the keys are the column name of the data frame associated
-            to *field_name* and the values are a list of values.
+            where *field_name* can be: sub, bus2sub, branch, bus, dcline, plant, plant, gencost_before, gencost_after, storage_gen, storage_StorageData and *data_dict* is a dictionary in which the keys are the column name of the data frame associated to *field_name* and the values are a list of values.
```

Comment:
Maybe we should break this long line into smaller ones.

Reply:
For sure

Reply:
Done